From 09407501712ba45dbcd3ae3c145e5fbeeb4f2527 Mon Sep 17 00:00:00 2001
From: Kosta Zertsekel <konszert@marvell.com>
Date: Sat, 3 Aug 2013 12:46:14 +0300
Subject: [PATCH 0921/1825] fix: iocc: alp: remove existing IO Sync Barrier WA

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit e25b9ded17adc6ba2137a6d59c40b9fcd91d8791

	Preparation to add new IO Sync Barrier WA using XOR engine.
	Rationale: increase readability.

Change-Id: Icd1e440c8f942e80bdcd68530c0b1bda4546039a
Signed-off-by: Kosta Zertsekel <konszert@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/2920
Reviewed-by: Igor Patrik <igorp@marvell.com>
Tested-by: Igor Patrik <igorp@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/mach-avantalp/core.c |   79 -----------------------------------------
 1 files changed, 0 insertions(+), 79 deletions(-)

diff --git a/arch/arm/mach-avantalp/core.c b/arch/arm/mach-avantalp/core.c
index 1b0f4b7..65dbbac 100644
--- a/arch/arm/mach-avantalp/core.c
+++ b/arch/arm/mach-avantalp/core.c
@@ -1281,92 +1281,13 @@ void alp_smp_cpu1_enable_wa(void)
 
 #ifdef CONFIG_ALP_IOCC_SYNC_BARRIER_WA
 
-static DEFINE_SPINLOCK(alp_io_sync_barrier_lock);
-void *io_sync_virt_addr_for_write;
-
 void dma_io_sync(void)
 {
-#if 1
-	udelay(1);
-#else
-	unsigned long flags;
-	u32 val = 0;
-	u32 *virt_uncached = (u32 *)(SZ_1K + CRYPT_ENG_VIRT_BASE(0));
-	u32 *virt_cached   = (u32 *)(SZ_1K + io_sync_virt_addr_for_write);
-
-	spin_lock_irqsave(&alp_io_sync_barrier_lock, flags);
-
-	/*
-	 * Write 0 through straight-forward path
-	 */
-	writel(0, virt_uncached);
-	dmb();
-
-	/*
-	 * Write 1 through CIB-SCU path
-	 */
-	*virt_cached = 1;
-	dmb();
-
-	/*
-	 * Wait till write through CIB-SCU path reaches SRAM
-	 */
-	do {
-		val = readl(virt_uncached);
-	} while (val != 1);
-
-	spin_unlock_irqrestore(&alp_io_sync_barrier_lock, flags);
-#endif
 }
 EXPORT_SYMBOL(dma_io_sync);
 
-static void alp_iocc_wa_setup_punit_win(ulong phys0, ulong phys1, MV_TARGET target)
-{
-	MV_AHB_TO_MBUS_DEC_WIN mbus_win;
-	MV_ADDR_WIN remap_win;
-	u32 win_num;
-
-	win_num = mvAhbToMbusWinAvailGet();
-	if (win_num >= MAX_AHB_TO_MBUS_REMAP_WINS || win_num == 0xffffffff) {
-		pr_err("%s: Error: No free PUnit window (win_num = 0x%x)\n",
-		       __func__, win_num);
-		return;
-	}
-
-	mbus_win.target			= target;
-	mbus_win.addrWin.baseLow	= phys0;
-	mbus_win.addrWin.baseHigh	= 0;
-	mbus_win.addrWin.size		= SZ_64K;
-	mbus_win.enable			= MV_TRUE;
-
-	if (mvAhbToMbusWinSet(win_num, &mbus_win) != MV_OK) {
-		pr_err("%s: Error: mvAhbToMbusWinSet(win_num = %d) failed\n",
-		       __func__, win_num);
-		return;
-	}
-
-	remap_win.baseLow  = phys1;	/* remap low base  */
-	remap_win.baseHigh = 0;		/* remap high base */
-
-	if (mvAhbToMbusWinRemap(win_num, &remap_win) == 0xffffffff) {
-		pr_err("%s: Error: mvAhbToMbusWinRemap(win_num = %d) failed\n",
-		       __func__, win_num);
-		return;
-	}
-}
-
 static void alp_iocc_wa_init(void)
 {
-	/*
-	 * Config PUnit windows
-	 */
-	alp_iocc_wa_setup_punit_win(IOCC_WA_WIN0_PHYS_BASE,
-				    CRYPT_ENG_PHYS_BASE(0), SDRAM_CS0);
-
-	/*
-	 * Save address for IOCC work-around write
-	 */
-	io_sync_virt_addr_for_write = (void *)IOCC_WA_WIN0_VIRT_BASE;
 }
 #else
 void dma_io_sync(void)
-- 
1.7.5.4

